src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -245,48 +245,186 @@ Or, ``source`` the script and run the tests manually::
 How to add a new controller?
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-If you want to add a new endpoint to the backend, you just need to add a
-class derived from ``BaseController`` decorated with ``ApiController`` in a
-Python file located under the ``controllers`` directory. The Dashboard module
-will automatically load your new controller upon start.
+A controller is a Python class that extends from the ``BaseController`` class
+and is decorated with either the ``@Controller`` or ``@ApiController``
+decorators. The Python class must be stored inside a Python file located under
+the ``controllers`` directory. The Dashboard module will automatically load
+your new controller upon start.
 
-For example create a file ``ping2.py`` under ``controllers`` directory with the
-following code::
+The ``@ApiController`` decorator is a specialization of the ``@Controller``
+decorator, and should be used for controllers that provide an API-like REST
+interface. For any other kinds of controllers the ``@Controller`` decorator
+should be used.
 
-  import cherrypy
-  from ..tools import ApiController, BaseController
+A controller has a URL prefix path associated that is specified in the
```

Comment:
...**an** URL...

Reply:
I think "**a** URL" is correct

Reply:
I think @p-na is right. URL sounds you-ar-l, see https://www.englishclub.com/pronunciation/a-an.htm or http://www.greyorgray.com/a-or-an.html.